PlayFabClientSDK/source/PluginManager.cs

```diff
@@ -1,44 +1,100 @@
+        }
+
+        /// <summary>
+        /// Sets a custom plugin.
+        /// If a plugin with specified contract and optional instance name already exists, it will be replaced with specified instance.
+        /// </summary>
+        /// <param name="plugin">The plugin instance.</param>
         /// <param name="contract">The app contract of plugin.</param>
-        /// <returns>The plugin implementation.</returns>
-        public IPlayFabPlugin GetPlugin(PluginContract contract)
+        /// <param name="instanceName">The optional plugin instance name. Instance names allow to have mulptiple plugins with the same contract.</param>
+        public void SetPlugin(IPlayFabPlugin plugin, PluginContract contract, string instanceName = "")
         {
-            if (!this.plugins.ContainsKey(contract))
+            Instance.SetPluginInternal(plugin, contract, instanceName);
+        }
+
+        private IPlayFabPlugin GetPluginInternal(PluginContract contract, string instanceName)
+        {
+            var key = new Tuple<PluginContract, string>(contract, instanceName);
+            if (!this.plugins.ContainsKey(key))
             {
-                // throw new ArgumentException("Plugin for contract is not set", nameof(contract));
-                return null;
+                IPlayFabPlugin plugin;
+                switch (contract)
+                {
+                    case PluginContract.PlayFab_Serializer:
+                        plugin = this.CreatePlayFabSerializerPlugin();
+                        break;
+                    case PluginContract.PlayFab_Transport:
+                        plugin = this.CreatePlayFabTransportPlugin();
+                        break;
+                    default:
+                        throw new ArgumentException("This contract is not supported", nameof(contract));
+                }
+
+                this.plugins[key] = plugin;
             }
 
-            return this.plugins[contract];
+            return this.plugins[key];
         }
 
-        /// <summary>
-        /// Sets a custom plugin.
-        /// </summary>
-        /// <param name="contract">The app contract of a custom plugin.</param>
-        /// <param name="plugin">The custom plugin implementation.</param>
-        public void SetPlugin(PluginContract contract, IPlayFabPlugin plugin)
+        private void SetPluginInternal(IPlayFabPlugin plugin, PluginContract contract, string instanceName)
+        {
+            if (plugin == null)
+            {
+                throw new ArgumentNullException(nameof(plugin), "Plugin instance cannot be null");
+            }
+
+            var key = new Tuple<PluginContract, string>(contract, instanceName);
+            this.plugins[key] = plugin;
+        }
+
+        private ITransportPlugin CreatePlayFabTransportPlugin()
+        {
+            var httpInterfaceType = typeof(IPlayFabHttp);
+            var types = typeof(PlayFabHttp).GetAssembly().GetTypes();
```

Comment:
typeof(X) here, X should be whatever class you're actually in here.
Formerly it was in PlayFabHttp, so it was typeof(PlayFabHttp)
This is PluginManager so it should be typeof(PluginManager)